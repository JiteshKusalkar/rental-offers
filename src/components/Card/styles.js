import styled from 'styled-components';

const Wrapper = styled.div`
  border-radius: 15px;
  overflow: hidden;
  box-shadow: 1px 1px 5px rgb(0 0 0 / 20%);
  transition: all 0.18s ease;

  :hover {
    transition: all 0.18s ease;
    transform: translateY(-1px);
    box-shadow: 0 1px 2px 0 rgb(60 64 67 / 30%),
      0 2px 6px 2px rgb(60 64 67 / 15%);
  }
`;
const Body = styled.div``;
const Img = styled.img``;

export { Body, Img, Wrapper };
