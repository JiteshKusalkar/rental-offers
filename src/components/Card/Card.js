import PropTypes from 'prop-types';
import { Body, Img, Wrapper } from './styles';

function Card({
  imageSrc,
  imageAlt,
  imageWidth = 350,
  imageHeight = 250,
  children,
  className,
}) {
  return (
    <Wrapper className={className}>
      <Img
        src={imageSrc}
        alt={imageAlt}
        width={imageWidth}
        height={imageHeight}
      />
      <Body>{children}</Body>
    </Wrapper>
  );
}

Card.propTypes = {
  children: PropTypes.node.isRequired,
  imageSrc: PropTypes.string,
  imageAlt: PropTypes.string,
  imageWidth: PropTypes.number,
  imageHeight: PropTypes.number,
  className: PropTypes.string,
};

export default Card;
