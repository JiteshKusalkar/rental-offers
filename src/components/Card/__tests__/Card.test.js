import { render } from '@testing-library/react';
import Card from '../Card';

describe('Card', () => {
  it('should render the component with valid props', () => {
    const props = {
      imageSrc: 'http://imageSrc',
      imageAlt: 'Image',
      imageWidth: 300,
      imageHeight: 300,
    };
    const { container } = render(<Card {...props}>Card content</Card>);

    expect(container).toMatchSnapshot();
  });

  it('should render the component with default props', () => {
    const props = {
      imageSrc: 'http://imageSrc',
      imageAlt: 'Image',
    };
    const { container } = render(<Card {...props}>Card content</Card>);

    expect(container).toMatchSnapshot();
  });
});
