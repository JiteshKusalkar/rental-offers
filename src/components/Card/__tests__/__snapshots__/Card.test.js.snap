// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render the component with default props 1`] = `
<div>
  <div
    class="sc-bdvvtL eaXtxm"
  >
    <img
      alt="Image"
      class="sc-dkPtRN"
      height="250"
      src="http://imageSrc"
      width="350"
    />
    <div
      class="sc-gsDKAQ"
    >
      Card content
    </div>
  </div>
</div>
`;

exports[`Card should render the component with valid props 1`] = `
<div>
  <div
    class="sc-bdvvtL eaXtxm"
  >
    <img
      alt="Image"
      class="sc-dkPtRN"
      height="300"
      src="http://imageSrc"
      width="300"
    />
    <div
      class="sc-gsDKAQ"
    >
      Card content
    </div>
  </div>
</div>
`;
